### ISSUE TITLE
Bun isn't respecting shebangs in vite, even with --bun flag

### ISSUE THREAD

**Issue by ReeceXW** [oven-sh/bun#6340]
### What version of Bun is running?

1.0.4+745b6b94ee56cad24d475799690cc9a89957d15b

### What platform is your computer?

Linux 5.15.123.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

```
bun create vite myapp
cd myapp
bunx --bun vite
```

### What is the expected behavior?

The process should be using bun

### What do you see instead?

The process is being run with node

### Additional information

If you change the shebang in vite/bin/vite.js to use bun, it works as expected

**Comment by jtmueller** [comment]
@Electroid I think the original title was correct. The `--bun` flag is supposed to override a shebang pointing to node, and instead it seems to have no effect. Vite still runs in a node process.

We want it to ignore the shebang, but instead it incorrectly honors it when using the `--bun` flag, so the original title of "Bun is respecting shebangs in vite, even with --bun flag" should be restored.

**Comment by jtmueller** [comment]
For example, running `bunx --bun vite` shows these processes:

```
83879 ttys002    0:00.01 bunx --bun vite
83880 ttys002    0:01.02 node /Users/me/dev/project/node_modules/.bin/vite
```

There seems to be no way to get Vite to actually run inside Bun short of directly targeting files in node_modules, so the claims that Bun fully supports Vite seem kind of specious.

**Comment by renhyl** [comment]
same issue here, I just realised that `bunx --bun vite` starts the vite with node process, confirmed by running `lsof -i :5173` COMMAND lists `node` rather than `bun`.

I don't know ZIG but I had a quick look, the issue might be here, or thereabouts, that entire logic is very confusing..

anyway, the `--bun` flag sets var 

https://github.com/oven-sh/bun/blob/c54844b30b2fff87728bc13da32fdc6f899b5b57/src/cli/run_command.zig#L759

then we have

https://github.com/oven-sh/bun/blob/c54844b30b2fff87728bc13da32fdc6f899b5b57/src/cli/run_command.zig#L780

and

https://github.com/oven-sh/bun/blob/c54844b30b2fff87728bc13da32fdc6f899b5b57/src/cli/run_command.zig#L805-L814

https://github.com/oven-sh/bun/blob/c54844b30b2fff87728bc13da32fdc6f899b5b57/src/cli/run_command.zig#L541

**Comment by paperclover** [comment]
I thought I had fixed this in the linked PR #9143 as i fixed parsing issues with --bun

But it turns out the issue described is not a bug in bun, but rather a misunderstanding in the output of lsof.

The shebang `/usr/bin/env node` looks for node, but with --bun, bun will place an exectuable named `node` in a temp directory, and add it to PATH. So bun is already using node. you can verify this by changing node_modules/.bin/vite to include a log
```
#!/usr/bin/env node
console.log(typeof Bun, process.argv0);
process.exit(1);
```

Notice how argv0 is still `node`

<img width="435" alt="image" src="https://github.com/oven-sh/bun/assets/24465214/6ce7fb92-7571-496a-ba79-d16269c24288">

The reason we do this intead of sniffing out the node shebang and replacing it, is because this PATH-based approach allows node to always be replaced with bun, for example a subprocess trying to spawn "node" will always hit bun in --bun mode, even if it is a non-javascript program like `which`

```
$ bunx --bun which node
/private/tmp/bun-node-1424a196f/node
```
